### PR TITLE
Continuos batching agent and documentation

### PR DESCRIPTION
**Pull request by baijumeswani**
Adds a github copilot agent for the continuous batching engine. And some documentation notes.

### CODE REVIEW

File: .github/agents/continuous-batching-engine.agent.md [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+2. Plan the requests, tokens, packed layout, and cache demand without advancing visible request progress.
+3. Reserve the complete cache growth required by the selected plan.
+4. Checkpoint every mutable request and sampler state that execution or sampling can change.
+5. Execute one model batch against the proposed cache and sequence layout.
+6. Stage logits processing, sampled tokens, completion, and ready events.
+7. Commit search state, cache ownership, request counters, status, and ready events in the defined order.
+8. On a recoverable failure, restore every request to the last committed boundary and release every reservation.
+9. Mark the Engine unhealthy when rollback or commit cannot prove a consistent state.
+
+Never approve a change that can leave request progress, sampler state, cache slots, block ownership, recurrent state, host token mirrors, or caller-visible events at different logical token boundaries.
```
Comment (tianleiwu) [inline .github/agents/continuous-batching-engine.agent.md]: Could you make the current recurrent-state support boundary explicit here? The dynamic Engine path constructs `SimpleDecoder`/`VarlenDecoderIO`, which binds packed inputs, logits, and paged KV cache, but it never creates or binds `RecurrentState`; that happens in the separate `DecoderOnly_State` generator path. `ScheduledRequests` therefore checkpoints search and sampler state only. Either document and validate that hybrid recurrent-state models are unsupported by dynamic batching, or add per-request recurrent-state binding, checkpoint, commit, and rollback before presenting this as an Engine invariant.

File: docs/paged_attention_engine.md [hunk 2]
```diff
@@ -0,0 +1,531 @@
...
+- `search.batch_size == 1`
+- `search.num_beams == 1`
+- At least one input token before the request is added to the engine
+
+The engine creates throughput by batching several independent requests, not by placing several sequence rows inside one request.
+
+The important request states are:
+
+```text
+Unassigned -> Assigned -> InProgress -> Completed
```
Comment (tianleiwu) [inline docs/paged_attention_engine.md:91]: The diagram makes `Remove()` look valid only after completion. In the implementation, `Request::Remove()` can also cancel `Assigned` or `InProgress` requests, and the dynamic scheduler erases them and releases any cache allocation immediately. Please show `Remove() -> Unassigned` from all engine-owned states, or annotate the diagram, so queued and active cancellation is part of the lifecycle contract.

**Review by tianleiwu — COMMENTED** [review]
The living guide and inline comments generally match the current implementation. I found two documentation contract gaps to clarify: the recurrent-state support boundary and removal transitions from active lifecycle states. Both are suggestions; this PR does not change runtime behavior.

**Review by tianleiwu — DISMISSED** [review]

**Review by tianleiwu — APPROVED** [review]
